base/distributed/clusterserialize.jl

```diff
@@ -88,8 +88,9 @@ function serialize(s::ClusterSerializer, g::GlobalRef)
     # Record if required and then invoke the default GlobalRef serializer.
     sym = g.name
     if g.mod === Main && isdefined(g.mod, sym)
-        v = getfield(Main, sym)
-         if (binding_module(Main, sym) === Main) && (s.anonfunc_id != 0)
+        if (binding_module(Main, sym) === Main) && (s.anonfunc_id != 0) &&
+            !startswith(string(sym), "#") # Anonymous functions are handled via FULL_GLOBALREF_TAG
```

Comment:
My understanding is: 

Merely checking for the symbol starting with `#` covers all hidden bindings under `Main`. As we anyway don't send type definitions, explicitly declared functions, etc, I don't see a problem. Anonymous functions are being handled as a special case in `serialize(s::AbstractSerializer, g::GlobalRef)` via `FULL_GLOBALREF_TAG`.